lib/everest.cpp

```diff
@@ -401,18 +411,32 @@ json Everest::call_cmd(const Requirement& req, const std::string& cmd_name, json
         res_future_status = res_future.wait_until(res_wait);
     } while (res_future_status == std::future_status::deferred);
 
-    json result;
+    CmdResult result;
     if (res_future_status == std::future_status::timeout) {
-        EVLOG_AND_THROW(EverestTimeoutError(fmt::format(
-            "Timeout while waiting for result of {}->{}()",
-            this->config.printable_identifier(connection["module_id"], connection["implementation_id"]), cmd_name)));
+        result.error = CmdResultError{
+            CmdEvent::Timeout,
+            fmt::format("Timeout while waiting for result of {}->{}()",
+                        this->config.printable_identifier(connection["module_id"], connection["implementation_id"]),
+                        cmd_name)};
     }
     if (res_future_status == std::future_status::ready) {
         result = res_future.get();
     }
     this->mqtt_abstraction->unregister_handler(cmd_topic, res_token);
 
-    return result;
+    if (result.error.has_value()) {
+        auto& error = result.error.value();
+        if (error.event == CmdEvent::HandlerException) {
+            throw HandlerException(fmt::format("{}", error.msg));
+        } else if (error.event == CmdEvent::Timeout) {
+            throw CmdTimeout(fmt::format("{}", error.msg));
+        }
+        throw CmdError(fmt::format("{}: {}", conversions::cmd_event_to_string(error.event), error.msg));
+    } else if (not result.result.has_value()) {
+        throw CmdError("Command did not return result");
+    } else {
+        return result.result.value();
+    }
```

Comment:
nit: reads a little better
```suggestion
if (result.error.has_value()) {
        const auto& error = result.error.value();
        const std::string err_msg = fmt::format("{}", error.msg);
    
        switch (error.event) {
            case CmdEvent::HandlerException:
                throw HandlerException(err_msg);
            case CmdEvent::Timeout:
                throw CmdTimeout(err_msg);
            default:
                throw CmdError(fmt::format("{}: {}", conversions::cmd_event_to_string(error.event), error.msg));
        }
    }
    
    if (!result.result.has_value()) {
        throw CmdError("Command did not return result");
    }
    
    return result.result.value();
```